Diagnostics/HealthChecker/Analyzer/Invoke-AnalyzerEngine.ps1

```diff
@@ -399,52 +399,80 @@
             -AnalyzedInformation $analyzedResults
     }
 
-    $displayWriteType2012 = "Yellow"
-    $displayWriteType2013 = "Yellow"
-    $displayValue2012 = "Unknown"
-    $displayValue2013 = "Unknown"
-
     if ($null -ne $osInformation.VcRedistributable) {
         Write-VerboseOutput("VCRedist2012 Testing value: {0}" -f [HealthChecker.VCRedistVersion]::VCRedist2012.value__)
         Write-VerboseOutput("VCRedist2013 Testing value: {0}" -f [HealthChecker.VCRedistVersion]::VCRedist2013.value__)
-        $vc2013Required = $exchangeInformation.BuildInformation.ServerRole -ne [HealthChecker.ExchangeServerRole]::Edge
-        $displayValue2012 = "Redistributable is outdated"
-        $displayValue2013 = "Redistributable is outdated"
 
         foreach ($detectedVisualRedistVersion in $osInformation.VcRedistributable) {
             Write-VerboseOutput("Testing {0} version id '{1}'" -f $detectedVisualRedistVersion.DisplayName, $detectedVisualRedistVersion.VersionIdentifier)
 
-            if ($detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2012) {
-                $displayValue2012 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
-                $displayWriteType2012 = "Green"
-            } elseif ($vc2013Required -and
-                $detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2013) {
-                $displayWriteType2013 = "Green"
-                $displayValue2013 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
+            if ($detectedVisualRedistVersion.DisplayName -like "Microsoft Visual C++ 2012*") {
+                $vcRedist2012Detected = $true
+                if ($detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2012) {
+                    $displayWriteType2012 = "Green"
+                    $displayValue2012 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
+                } else {
+                    $displayWriteType2012 = "Yellow"
+                    $displayValue2012 = "Redistributable is outdated"
+                }
+            }
+
+            if ($detectedVisualRedistVersion.DisplayName -like "Microsoft Visual C++ 2013*") {
+                $vcRedist2013Detected = $true
+                if ($detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2013) {
+                    $displayWriteType2013 = "Green"
+                    $displayValue2013 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
+                } else {
+                    $displayWriteType2013 = "Yellow"
+                    $displayValue2013 = "Redistributable is outdated"
+                }
             }
         }
     }
 
+    if (!($vcRedist2012Detected)) {
```

Comment:
There isn't a point of doing this as it is the same as setting the default. Let's not make it so complicated and add it back to how it was with the default.

Reply:
Done.